src/libsyntax_pos/symbol.rs

```diff
@@ -16,6 +17,94 @@ use std::hash::{Hash, Hasher};
 use crate::hygiene::SyntaxContext;
 use crate::{Span, DUMMY_SP, GLOBALS};
 
+symbols! {
+    // After modifying this list adjust `is_special`, `is_used_keyword`/`is_unused_keyword`,
+    // this should be rarely necessary though if the keywords are kept in alphabetic order.
+    Keywords {
+        // Special reserved identifiers used internally for elided lifetimes,
+        // unnamed method parameters, crate root module, error recovery etc.
+        Invalid,            "",
+        PathRoot,           "{{root}}",
+        DollarCrate,        "$crate",
+        Underscore,         "_",
+
+        // Keywords that are used in stable Rust.
+        As,                 "as",
```

Comment:
Nit: Could you use something that's not a comma as a separator here, like `:` or `=`.

Reply:
Done.